src/AdoNetCore.AseClient/Internal/SchemaTableBuilder.cs

```diff
@@ -163,28 +168,45 @@ private void TryLoadKeyInfo(DataTable table, string baseTableNameValue, string b
 
                 try
                 {
+                    var columnMetadata = new Dictionary<string, ColumnMetadata>();
                     using (var keyInfoDataReader = command.ExecuteReader())
                     {
                         while (keyInfoDataReader.Read())
                         {
-                            var keyColumnName = keyInfoDataReader["COLUMN_NAME"].ToString();
-                            var keySchemaName = keyInfoDataReader["TABLE_SCHEMA"].ToString();
-                            var keyCatalogName = keyInfoDataReader["TABLE_CATALOG"].ToString();
+                            var key = keyInfoDataReader["COLUMN_NAME"].ToString();
+
+                            // Check for duplicate name and if we found any then give it a miss
+                            if (columnMetadata.ContainsKey(key))
+                            {
+                                keyInfoDataReader.Close();
+                                return;
+                            }
+
+                            columnMetadata.Add(key, new ColumnMetadata
+                            {
+                                Name = key,
+                                Schema = keyInfoDataReader["TABLE_SCHEMA"].ToString(),
+                                Catalog = keyInfoDataReader["TABLE_CATALOG"].ToString(),
+                                IsKey = (bool) keyInfoDataReader["PRIMARY_KEY"],
+                                IsUnique = (bool) keyInfoDataReader["UNIQUE"]
+                            });
+                        }
+                    }
 
-                            foreach (DataRow row in table.Rows)
+                    foreach (var metadata in columnMetadata.Values)
```

Comment:
This introduces an additional loop through all the metadata? Why not do it in place like it was before?

Reply:
Because, you're need to check against the dictionary of distinct column name.